samza-api/src/main/java/org/apache/samza/checkpoint/CheckpointV2.java

```diff
@@ -37,17 +36,18 @@
 
   private final CheckpointId checkpointId;
   private final Map<SystemStreamPartition, String> inputOffsets;
-  private final Map<String, List<StateCheckpointMarker>> stateCheckpointMarkers;
+  private final Map<String, Map<String, String>> stateCheckpointMarkers;
 
   /**
    * Constructs the checkpoint with separated input and state offsets
    * @param checkpointId CheckpointId associated with this checkpoint
    * @param inputOffsets Map of Samza system stream partition to offset of the checkpoint
-   * @param stateCheckpointMarkers Map of local state store names and StateCheckpointMarkers for each state backend system
+   * @param stateCheckpointMarkers Map of state backend factory name to map of local state store names
+   *                               to StateCheckpointMarkers
```

Comment:
s/StateCheckpointMarkers/state checkpoint markers everywhere since its not a class anymore.